src/dmd/root/file.d

```diff
@@ -96,17 +92,14 @@ nothrow:
             size_t sz = bufIncrement;
 
             if (!_ref)
-                .free(buffer);
+                mem.xfree(buffer);
```

Comment:
why change from malloc/free to mem?

Reply:
Presumably to play nice with `-lowmem`.

Reply:
"presumably" seems rather vague.

Reply:
See commit msges - the dtor was using `mem.xfree()` (presumably to free the `mem`-managed OutBuffers after taking their ownership), but `read()` C malloc/realloc/free (and so call sites had to either take ownership of the file buffer or free it manually).